ttkHyperlinkLabel.py

```diff
@@ -1,26 +1,40 @@
+"""
+A clickable ttk label for HTTP links.
+
+In addition to standard ttk.Label arguments, takes the following arguments:
+  url: The URL as a string that the user will be sent to on clicking on
+  non-empty label text. If url is a function it will be called on click with
+  the current label text and should return the URL as a string.
+  underline: If True/False the text is always/never underlined. If None (the
+  default) the text is underlined only on hover.
+  popup_copy: Whether right-click on non-empty label text pops up a context
+  menu with a 'Copy' option. Defaults to no context menu. If popup_copy is a
+  function it will be called with the current label text and should return a
+  boolean.
+
+May be imported by plugins
+"""
 import sys
 import tkinter as tk
 import webbrowser
-from tkinter import font as tkFont
+from tkinter import font as tk_font
 from tkinter import ttk
+from typing import TYPE_CHECKING, Any, Optional
 
 if sys.platform == 'win32':
     import subprocess
     from winreg import HKEY_CLASSES_ROOT, HKEY_CURRENT_USER, CloseKey, OpenKeyEx, QueryValueEx
 
-# A clickable ttk Label
-#
-# In addition to standard ttk.Label arguments, takes the following arguments:
-#   url: The URL as a string that the user will be sent to on clicking on non-empty label text. If url is a function it will be called on click with the current label text and should return the URL as a string.
-#   underline: If True/False the text is always/never underlined. If None (the default) the text is underlined only on hover.
-#   popup_copy: Whether right-click on non-empty label text pops up a context menu with a 'Copy' option. Defaults to no context menu. If popup_copy is a function it will be called with the current label text and should return a boolean.
-#
-# May be imported by plugins
+if TYPE_CHECKING:
+    def _(x: str) -> str: ...
 
 
-class HyperlinkLabel(sys.platform == 'darwin' and tk.Label or ttk.Label, object):
+# FIXME: Split this into multi-file module to separate the platforms
+class HyperlinkLabel(sys.platform == 'darwin' and tk.Label or ttk.Label, object):  # type: ignore
+    """Clickable label for HTTP links."""
 
-    def __init__(self, master=None, **kw):
+    # NB: do **NOT** try `**kw: Dict`, it causes more trouble than it's worth.
+    def __init__(self, master: Optional[tk.Tk] = None, **kw) -> None:
```

Comment:
kw: Any?

Reply:
That I did not try, but I'll be unsurprised if mypy still insists that then doesn't match with what typeshed thinks is going on with actual uses.  At that stage it might as well stay untyped ?

The 'make mypy happy' throughout this was mostly get satisfy the flake8 "must have at least 75% (default) types coverage".

Reply:
I ought not respond to PR comments just before bed.

Yup, `Any` works, and at least makes it explicit.  Also used in the `__setitem__()` signature now.